### PR TITLE
ZEPPELIN-598 ] Dynamic loading for Interpreter and API

### PR DESCRIPTION
**Pull request by cloverhearts**
### What is this PR for?

Use of external libraries that are included in the Roadmap,
The interpreter needs related to dynamic loading capabilities.
### What type of PR is it?

Feature

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-598

### How should this be tested?
1. Run to `mvn clean package -DskipTests`
2. Run to `mvn clean package -DskipTests -pl zeppelin-server --am`
3. run zeppelin.
4. move to interpreter menu on web ui.
   - check empty interpreter
5. call restapi
   
```
POST http://localhost:8080/api/interpreter/load/md/md
{"artifact":"org.apache.zeppelin:zeppelin-markdown:0.6.2", "className":"org.apache.zeppelin.markdown.Markdown"}
```

6. check interpreter menu to create interpreter `md` on web ui.
7. add interpreter setting `md`
8. run paragraph `md`
```
%md
Hello markdown
```
9. call unload restapi
```
DELETE http://localhost:8080/api/interpreter/unload/md2/md2/org.apache.zeppelin.markdown.Markdown
```
10. check delete `md` on web ui
11. restart to paragraph for `md` 
     if result is error status then well working.

### Screenshots (if appropriate)
[Click to link for screenshot](https://cloud.githubusercontent.com/assets/10525473/22174189/1b2cce6c-df8d-11e6-975f-333a502079ea.gif)

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? we have had already.

### CODE REVIEW

**Comment by cloverhearts** [comment]
LOG FILE Limit 4MB Over

File: conf/zeppelin-env.cmd.template [hunk 1]
```diff
@@ -37,6 +37,7 @@ REM set ZEPPELIN_NOTEBOOK_S3_KMS_KEY_REGION    REM AWS KMS key region
 REM set ZEPPELIN_IDENT_STRING   		REM A string representing this instance of zeppelin. $USER by default.
 REM set ZEPPELIN_NICENESS       		REM The scheduling priority for daemons. Defaults to 0.
 REM set ZEPPELIN_INTERPRETER_LOCALREPO         REM Local repository for interpreter's additional dependency loading
+REM set ZEPPELIN_INTERPRETER_DOWNLOAD_DIR       REM Path for interpreter download
```
Comment (felixcheung) [inline conf/zeppelin-env.cmd.template]: could you please align the 2nd REM?

File: conf/zeppelin-env.sh.template [hunk 2]
```diff
@@ -38,6 +38,7 @@
 # export ZEPPELIN_IDENT_STRING   		# A string representing this instance of zeppelin. $USER by default.
 # export ZEPPELIN_NICENESS       		# The scheduling priority for daemons. Defaults to 0.
 # export ZEPPELIN_INTERPRETER_LOCALREPO         # Local repository for interpreter's additional dependency loading
+# export ZEPPELIN_INTERPRETER_DOWNLOAD_DIR          # Path for interpreter download
```
Comment (felixcheung) [inline conf/zeppelin-env.sh.template]: could you please align the `# Path`

File: conf/zeppelin-env.cmd.template (hunk 1 above)
Comment (cloverhearts) [inline conf/zeppelin-env.cmd.template]: @felixcheung Okay :) I will fix it. Thank you :)

File: conf/zeppelin-env.sh.template (hunk 2 above)
Comment (cloverhearts) [inline conf/zeppelin-env.sh.template]: Okay :)

File: conf/zeppelin-env.cmd.template (hunk 1 above)
Comment (cloverhearts) [inline conf/zeppelin-env.cmd.template]: fixed :)

File: conf/zeppelin-env.sh.template (hunk 2 above)
Comment (cloverhearts) [inline conf/zeppelin-env.sh.template]: fixed :)

**Comment by jongyoul** [comment]
@cloverhearts StaticInitialization from `Interpreter.register` is deprecated thus you cannot init new interpreter by `Class.forName`. You have to use new way to install new interpreter.

**Pull request closed without merging**